File: ios/Demo/MekuriDemo/DemoTitlePage.swift
import SwiftUI

/// The cover: one full panel with the title over a skyline, and a small
/// inset panel for the broadcast.
struct DemoTitlePage: View {
    let scale: CGFloat

    var body: some View {
        DemoPanelGrid(frames: [CGRect(x: 0, y: 0, width: 1, height: 1)]) { _ in
            DemoPanel {
                GeometryReader { proxy in
                    ZStack {
                        DemoMoon()
                            .frame(width: proxy.size.width * 0.42, height: proxy.size.width * 0.42)
                            .position(x: proxy.size.width * 0.70, y: proxy.size.height * 0.24)
                        DemoSkyline()
                            .frame(height: proxy.size.height * 0.30)
                            .frame(maxHeight: .infinity, alignment: .bottom)
                        VStack(spacing: 6 * self.scale) {
                            Text("Chapter One")
                                .font(DemoInk.narration(16 * self.scale))
                                .italic()
                            Text("MEKURI")
                                .font(DemoInk.title(58 * self.scale))
                                .tracking(4 * self.scale)
                            Text("the page turns the way paper does")
                                .font(DemoInk.narration(14 * self.scale))
                        }
                        .foregroundStyle(DemoInk.ink)
                        .padding(.horizontal, 20 * self.scale)
                        .padding(.vertical, 14 * self.scale)
                        .background(Rectangle().fill(DemoInk.paper).stroke(DemoInk.ink, lineWidth: DemoInk.panelStroke))
                        .position(x: proxy.size.width * 0.5, y: proxy.size.height * 0.60)
                        DemoBroadcastPanel(size: 13 * self.scale)
                            .padding(10 * self.scale)
                            .background(Rectangle().fill(DemoInk.paper).stroke(DemoInk.ink, lineWidth: DemoInk.panelStroke))
                            .frame(maxWidth: .infinity, maxHeight: .infinity, alignment: .bottomLeading)
                            .padding(-DemoInk.panelStroke)
                    }
                }
            }
        }
    }
}

File: android/demo/src/main/kotlin/studio/anicanon/mekuri/demo/DemoTitlePage.kt
package studio.anicanon.mekuri.demo

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.BoxWithConstraints
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.text.BasicText
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.BiasAlignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.geometry.Rect
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp

/**
 * The cover: one full panel with the title over a skyline, and a small inset
 * panel for the broadcast.
 */
@Composable
fun DemoTitlePage(scale: Float) {
    DemoPanelGrid(frames = listOf(Rect(0f, 0f, 1f, 1f))) {
        DemoPanel(Modifier.fillMaxSize()) {
            BoxWithConstraints(Modifier.fillMaxSize()) {
                val width = this.maxWidth
                val height = this.maxHeight
                DemoMoon(
                    Modifier
                        .size(width * 0.42f)
                        .align(BiasAlignment(horizontalBias = 0.4f, verticalBias = -0.52f)),
                )
                DemoSkyline(
                    Modifier
                        .fillMaxWidth()
                        .height(height * 0.30f)
                        .align(Alignment.BottomStart),
                )
                Column(
                    Modifier
                        .align(verticalBias(0.60f))
                        .background(DemoInk.Paper)
                        .border(DemoInk.PanelStroke, DemoInk.Ink)
                        .padding(horizontal = (20 * scale).dp, vertical = (14 * scale).dp),
                    horizontalAlignment = Alignment.CenterHorizontally,
                    verticalArrangement = Arrangement.spacedBy((6 * scale).dp),
                ) {
                    BasicText("Chapter One", style = DemoInk.narration(16 * scale))
                    BasicText(
                        "MEKURI",
                        style = DemoInk.title(58 * scale).copy(textAlign = TextAlign.Center),
                    )
                    BasicText(
                        "the page turns the way paper does",
                        style = DemoInk.narration(14 * scale).copy(textAlign = TextAlign.Center),
                    )
                }
                Box(
                    Modifier
                        .align(Alignment.BottomStart)
                        .background(DemoInk.Paper)
                        .border(DemoInk.PanelStroke, DemoInk.Ink)
                        .padding((10 * scale).dp),
                ) {
                    DemoBroadcastPanel(size = 13 * scale)
                }
            }
        }
    }
}
